ragtime.jdbc/src/ragtime/jdbc.clj

```diff
@@ -17,10 +17,18 @@
                         [[:id "varchar(255)"]
                          [:created_at "varchar(32)"]]))
 
-(defn- ensure-migrations-table-exists [db-spec migrations-table]
-  (try
-    (sql/execute! db-spec [(migrations-table-ddl migrations-table)])
-    (catch SQLException _)))
+(defn- table-exists?
+  [connection table-name]
+  (when connection
+    (let [results (.getTables (.getMetaData connection) nil nil table-name nil)
+          n (.next results)]
+      (and n (= (.getString results 3) table-name)))))
+
+(defn- ensure-migrations-table-exists
+  [{:keys [connection] :as db-spec} migrations-table]
+  ;; basically everything should be upper-cased except postgres
+  (when-not (table-exists? connection (.toUpperCase migrations-table))
```

Comment:
Mmm. Yeah, that just seemed sort of expensive to do on each migration. Something else I was thinking about was setting some sort of context for migrations to be applied, during which we would keep track of the existence of the migration table in a dynamic variable. I assumed you'd thought of that before and chosen not to pursue it for some reason.